src/lib/upgradePackageDefinitions.ts

```diff
@@ -11,6 +11,77 @@ import { pickBy } from './pick'
 import queryVersions from './queryVersions'
 import upgradeDependencies from './upgradeDependencies'
 
+/**
+ * check if in peer violation
+ *
+ * @returns
+ */
+const checkIfInPeerViolation = (
+  currentDependencies: Index<VersionSpec>,
+  filteredUpgradedDependencies: Index<VersionSpec>,
+  upgradedPeerDependencies: Index<Index<VersionSpec>>,
+) => {
+  const upgradedDependencies = { ...currentDependencies, ...filteredUpgradedDependencies }
+  const upgradedDependenciesVersions = Object.fromEntries(
+    Object.entries(upgradedDependencies).map(([packageName, versionSpec]) => {
+      return [packageName, semver.minVersion(versionSpec)?.version ?? versionSpec]
+    }),
+  )
+  const filteredUpgradedPeerDependencies = { ...upgradedPeerDependencies }
+  let wereUpgradedDependenceFiltered = false
```

Comment:
Let's call variable this `violated` to be consistent with the function name.

Reply:
done